File: cap6/teste/teste.js
const fs = require('fs');

const teste = (index) => {
  try {
    const data = fs.readFileSync('./teste.txt', 'utf8');

    const lowerCased = data.toLowerCase();

    const withoutComma = lowerCased.replace(/[\,|\.]/, '');

    const splitted = withoutComma.split(' ');

    const replacedMap = splitted.map((word) => {
      if (word[0] === 'a') return 1;
      if (word[0] === 'b') return 2;
      if (word[0] === 'c') return 3;
      if (word[0] === 'd') return 4;
      if (word[0] === 'e') return 5;
      if (word[0] === 'f') return 6;
      if (word[0] === 'g') return 7;
      if (word[0] === 'h') return 8;
      if (word[0] === 'i') return 9;
      return 10;
    });

    const removedNumberTen = replacedMap.filter((number) => number != 10);

    const multipliedForTen = removedNumberTen.map((number) => number * index);

    const resultSum = multipliedForTen.reduce(
      (accumulator, currentValue) => accumulator + currentValue,
      0,
    );

    return resultSum;
  } catch (error) {
    console.log('Deu erro!');
  }
};

const t1 = Date.now();
for (let index = 0; index < 50_000; index++) {
  teste(index);
}
const t2 = Date.now();
const timer = t2 - t1;

console.log(`tempo: ${timer}ms`);
// tempo: 34_197ms

File: cap6/teste/teste.exs
defmodule Teste do
  def call(index) do
    t1 = System.monotonic_time(:millisecond)

    case File.read("./teste.txt") do
      {:ok, content} ->
        content
        |> lower_cased()
        |> without_comma()
        |> splinted()
        |> replaced_map()
        |> removed_number_ten()
        |> multiplied_for_ten(index)
        |> result_sum()
        |> IO.inspect()

      _ ->
        IO.puts("Deu erro!")
    end

    t2 = System.monotonic_time(:millisecond)
    t2 - t1
  end

  defp lower_cased(text), do: String.downcase(text)

  defp without_comma(text), do: String.replace(text, [",", "."], "")

  defp splinted(text), do: String.split(text)

  defp replaced_map(text) do
    text
    |> Enum.map(fn word ->
      cond do
        String.first(word) == "a" -> 1
        String.first(word) == "b" -> 2
        String.first(word) == "c" -> 3
        String.first(word) == "d" -> 4
        String.first(word) == "e" -> 5
        String.first(word) == "f" -> 6
        String.first(word) == "g" -> 7
        String.first(word) == "h" -> 8
        String.first(word) == "i" -> 9
        true -> 10
      end
    end)
  end

  defp removed_number_ten(list) do
    list |> Enum.filter(fn number -> number != 10 end)
  end

  defp multiplied_for_ten(list, index) do
    list |> length() |> IO.puts()
    list |> Enum.map(fn number -> number * index end)
  end

  defp result_sum(list) do
    list
    |> Enum.reduce(0, fn number, accumulator ->
      accumulator + number
    end)
  end
end

# 5
# 13
# 15
# 19
# 20
# 21
# 20

# 21

total = Enum.reduce(1..25_000, 0, fn index, acc -> acc + Teste.call(index) end)

IO.puts("total: #{total}")
# 204041ms
